cirq/api/google/v2/result.proto

```diff
@@ -0,0 +1,103 @@
+syntax = "proto3";
+
+import "cirq/api/google/v2/program.proto";
+
+package cirq.api.google.v2;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "ResultProto";
+option java_multiple_files = true;
+
+// The overall results of running a Program.
+message Result {
+  // The results for each ParameterSweep. These will be in the same order
+  // as the parameter_sweeps repeated field in the RunContext that generated
+  // these results.
+  repeated SweepResult sweep_results = 1;
+}
+
+// The measurement results for a particular ParameterSweep.
+message SweepResult {
+  // The total number of repetitions that were performed for this sweep.
+  // This is reported in order to make it possible to decode the bytes
+  // in the measurement results.
+  int32 repetitions = 1;
+
+  // A list of measurement keys (string) along with the qubits that have been
+  // measured. The size of the measurement key is the total number of
+  // measurements in the list of operations for the Program. The measurement
+  // keys are all unique.
+  repeated MeasurementKey measurement_keys = 2;
+
+  // The results along with the parameters that generated these results.
+  // These represent the expanded parameters defined int he ParameterSweep
+  // which this SweepResult corresponds to.
+  repeated ParameterizedResult parameterized_results = 3;
+}
+
+// The parameters used to generate result along with the results for this
+// set of parameters.
+message ParameterizedResult {
+  // The parameter dict that was used when generating these results.
+  ParameterDict params = 1;
+
+  // The measurement results. This is a packed representation of all of the
+  // measurements for fixed set of parameters (specified by params above)
+  // across all of the repetitions for the set of parameters given the
+  // params field.
+  //
+  // The result of a measurement gate is a list of bits, one for each qubit
+  // that has been measured. Every Program has a fixed set of measurement gates.
+  // These gates are labeled by a key (this key is unique across the Program).
+  // The results of a single run of the program should consist of bits for
+  // each of the measurement gates. To order these bits we use the
+  // measurement_keys field of the SweepResult. This is an ordered list of
```

Comment:
The measurement_keys are ordered lexiographically?

Reply:
No, there would be an argument order specific to the gate type. Similar to the argument order, but just specific to outputs instead of inputs.